modules/nwtc-library/src/NWTC_Num.f90

```diff
@@ -2246,6 +2250,49 @@ FUNCTION EulerExtractR16(M) result(theta)
             
       
    END FUNCTION EulerExtractR16
+
+!=======================================================================
+!> 
+   FUNCTION EulerConstructZYXR8(theta) result(M)
+   
+      ! this function creates a rotation matrix, M, from a 3-2-1 rotation
+      ! sequence of the 3 Euler angles, theta_z, theta_y, and theta_x, in radians.
+      ! M represents a change of basis (from global to local coordinates; 
+      ! not a physical rotation of the body). it is the inverse of EulerExtract
```

Comment:
This would be the inverse of EulerExtractZYX(), if such a routine is written.  EulerExtract uses an XYZ rotation sequence.